### PR TITLE
feat: fix types according to https://opendocs.alipay.com/mini/api/vzt2xm

### PR DESCRIPTION
**Pull request by guoshencheng**
fix types according to https://opendocs.alipay.com/mini/api/vzt2xm

### CODE REVIEW

File: types/api/device/clipboard.d.ts [hunk 1]
```diff
@@ -19,6 +19,20 @@ declare namespace my {
     complete?(): void;
   }
 
+  interface ISetClipboardOptions {
```
Comment (guoshencheng) [inline types/api/device/clipboard.d.ts:22]: 参照文档 [https://opendocs.alipay.com/mini/api/klbkbp](https://opendocs.alipay.com/mini/api/klbkbp)

File: types/api/device/settings.d.ts [hunk 2]
```diff
@@ -2,7 +2,7 @@
  * @file 设置
  */
 declare namespace my {
-  type SettingScopeList = 'scope.userInfo' | 'scope.location' | 'scope.album' | 'scope.camera' | 'scope.audioRecord';
+  type SettingScopeList = 'userInfo' | 'location' | 'album' | 'camera' | 'audioRecord';
```
Comment (guoshencheng) [inline types/api/device/settings.d.ts:5]: 参照文档[https://opendocs.alipay.com/mini/api/xmk3ml](https://opendocs.alipay.com/mini/api/xmk3ml)

File: types/api/location.d.ts [hunk 3]
```diff
@@ -62,7 +62,7 @@ declare namespace my {
      * 国家(type>0生效)
      */
     readonly country?: string;
-    readonly bearing: string;
+    readonly bearing?: string;
```
Comment (guoshencheng) [inline types/api/location.d.ts:65]: 参照文档[https://opendocs.alipay.com/mini/api/mkxuqd](https://opendocs.alipay.com/mini/api/mkxuqd)

File: types/api/media/index.d.ts [hunk 4]
```diff
@@ -176,6 +176,12 @@ declare namespace my {
    */
   function saveImage(options: ISaveImageOptions): void;
 
+  /**
```
Comment (guoshencheng) [inline types/api/media/index.d.ts:179]: 参照文档[https://opendocs.alipay.com/mini/api/ehndze](https://opendocs.alipay.com/mini/api/ehndze)

File: types/api/ui/feedback.d.ts [hunk 5]
```diff
@@ -39,10 +39,15 @@ declare namespace my {
    */
   function showToast(options: IShowToastOptions): void;
 
+  interface IHideToastOptions {
```
Comment (guoshencheng) [inline types/api/ui/feedback.d.ts:42]: refer [https://opendocs.alipay.com/mini/api/iygd4e](https://opendocs.alipay.com/mini/api/iygd4e)

File: types/api/ui/refresh.d.ts [hunk 6]
```diff
@@ -2,13 +2,23 @@
  * @file 下拉刷新
  */
 declare namespace my {
+  interface IStartPullDownRefresh {
```
Comment (guoshencheng) [inline types/api/ui/refresh.d.ts]: refer [https://opendocs.alipay.com/mini/api/pmhkbb](https://opendocs.alipay.com/mini/api/pmhkbb)

File: types/api/ui/scroll.d.ts [hunk 7]
```diff
@@ -4,9 +4,29 @@
 declare namespace my {
   interface IPageScrollToOptions {
     /**
-     * 滚动到页面的目标位置，单位 px
+     * 滚动到页面的目标位置，单位 px。使用 my.pageScrollTo 跳转小程序顶部时，必须将 scrollTop 值设为大于 0，方可实现跳转
```
Comment (guoshencheng) [inline types/api/ui/scroll.d.ts:7]: refer [https://opendocs.alipay.com/mini/api/scroll](https://opendocs.alipay.com/mini/api/scroll)

File: types/api/location.d.ts (hunk 3 above)
Comment (yibuyisheng) [reply types/api/location.d.ts:65]: 看了下文档，好像没有 bearing 参数了？

File: types/api/ui/navigator.d.ts [hunk 8]
```diff
@@ -104,9 +104,17 @@ declare namespace my {
...
+    success?: () => void;
+    fail?: () => void;
+    complete?: () => void;
+  }
   /**
    * 隐藏TitleBar上的返回首页图标，和通用菜单中的“返回首页”功能。
    * 返回首页功能出现时机：当用户启动小程序，若直接进入的页面不是小程序的首页，则会在左上角出现返回首页icon，若用户继续在页面中进入下一级页面，则在右上角更多菜单中，会出现“返回首页”功能。
    */
-  function hideBackHome(): void;
+  function hideBackHome(options?: IHideBackHomeOptions): void;
```
Comment (yibuyisheng) [inline types/api/ui/navigator.d.ts]: 文档上面没写有这些参数：https://opendocs.alipay.com/mini/api/ui-navigate

File: types/api/ui/refresh.d.ts (hunk 6 above)
Comment (yibuyisheng) [reply types/api/ui/refresh.d.ts]: IStartPullDownRefresh -> IStartPullDownRefreshOptions

File: types/api/ui/refresh.d.ts [hunk 9]
```diff
@@ -2,13 +2,23 @@
...
+    success?: () => void;
+    fail?: () => void;
+    complete?: () => void;
+  }
+  interface IStopPullDownRefresh {
```
Comment (yibuyisheng) [inline types/api/ui/refresh.d.ts]: IStopPullDownRefresh -> IStopPullDownRefreshOptions

File: types/api/location.d.ts (hunk 3 above)
Comment (guoshencheng) [reply types/api/location.d.ts:65]: 真机优惠返回这个值，因此定义成了可选

**Pull request merged**